bundles/org.openhab.binding.modbus/src/test/java/org/openhab/binding/modbus/internal/profiles/ModbusGainOffsetProfileTest.java

```diff
@@ -172,10 +172,7 @@ private void testOnUpdateFromHandlerGeneric(String preGainOffset, String gain, O
         // Workaround for errors like "java.lang.UnsupportedOperationException: °C is non-linear, cannot convert"
         if (expectedStateUpdateTowardsItem instanceof QuantityType<?>) {
             assertTrue(actualStateUpdateTowardsItem instanceof QuantityType<?>);
-            assertEquals(((QuantityType<?>) expectedStateUpdateTowardsItem).getUnit(),
-                    ((QuantityType<?>) actualStateUpdateTowardsItem).getUnit());
-            assertEquals(((QuantityType<?>) expectedStateUpdateTowardsItem).toBigDecimal(),
-                    ((QuantityType<?>) actualStateUpdateTowardsItem).toBigDecimal());
+            assertEquals(expectedStateUpdateTowardsItem, actualStateUpdateTowardsItem);
```

Comment:
Interesting to see the complicated equals check is no longer needed. I suppose it was remnant from time (during development) when I tried to make the profile support gain/offset for any QuantityType. This caused unexpected temperature unit errors like indicated in the comment above.

If the tests pass, it should be fine. 

One change suggestion, can you remove now unnecessary conditional and the obsolete comment on "workaround"

Reply:
>  unnecessary conditional

Which one do you think is unnecessary?

Reply:
OK now I understand, I've created https://github.com/openhab/openhab-addons/pull/10670 to clean it up. :+1: